### DISCUSSION TITLE
Implementation websockets in adonisjs 5

### DISCUSSION THREAD

**Discussion by adnanerlansyah403** [adonisjs/core#4292] [Help]
Is there any documentation that explains about how to setup websocket in adonisjs 5 ? but not for View template. I want to use it on NuxtJS 3. Please let me know if there's a way to do it. I still struggling with this feat, because there are only a few topics that discuss websocket and that too only in the old version or 4.

I already setup it on client side, this is my code on the socket.ts file:
```
// utils/socket.ts

export const startSocket = () => {
    
    const websocket = new WebSocket("ws://127.0.0.1:3333/");
    console.log("Starting Connection to Websocket Server");

    websocket.onopen = (res) => {
        console.log("Successfully connected");
    };

    websocket.onmessage = (res) => {
    };

    websocket.onclose = () => {
        console.log("Server closed");
    };
    
    websocket.onerror = (error: any) => {
        console.error('WebSocket error:', error)
    }
};
```
And when I hit that endpoint, I just got the issue like this:
```
WebSocket connection to 'ws://127.0.0.1:3333/' failed: 
```

**Comment by MaximeMRF** [comment]
Hello @adnanerlansyah403 

Yes there is a [cookbook](https://docs.adonisjs.com/cookbooks/socketio-with-adonisjs) to use `socket.io` with AdonisJs v5

**Reply by adnanerlansyah403** [reply]
![codetemplate](https://github.com/adonisjs/core/assets/88386921/1c74e14e-6e30-49a7-a7c7-dc11266b2400)
![consolelogterminal](https://github.com/adonisjs/core/assets/88386921/e62953c0-a08b-49f4-8b99-5c744e55b5c9)
![codesetupsocket](https://github.com/adonisjs/core/assets/88386921/939a730b-81c4-4586-b5d7-cdbf3f1dfd92)


It still not working when I already tried from the tutorial, the data of console didn't appears when I opened the browser. Oh yeah btw I tried this on server side or from pure adonisjs5

**Comment by adnanerlansyah403** [comment]
Anyone can help me out with this issue ? I need complete my project as fast as I can for my homework

**Reply by MohamedAljoke** [reply]
ah check this one
#4406

**Comment by zecar** [comment]
i think you can guide yourself with this: https://v5-docs.adonisjs.com/cookbooks/socketio-with-adonisjs

you can replace the socket.io server with websocket

**Reply by MaximeMRF** [reply]
I have already mention this cookbook
